Framework/Algorithms/src/CreateWorkspace.cpp

```diff
@@ -144,16 +161,15 @@ void CreateWorkspace::exec() {
   }
 
   // Create the OutputWorkspace
+  auto storageMode = Parallel::fromString(getProperty("ParallelStorageMode"));
+  IndexInfo indexInfo(nSpec, storageMode, communicator());
```

Comment:
Hmm, that is an interesting idea!

```cpp
alg.setProperty("OutputWorkspace", "ws", StorageMode::Distributed);
```

I think I kind of like that, but we need to think about it more. Implementing this might mean some significant effort. Also, it is not clear how the communicator would be injected? How do you actually create the `IndexInfo`? You simply query the output workspace property for the storage mode? But then how is this better then having a separate property?